### PR TITLE
Upgrade to ML.NET version 1.0.0

### PR DESCRIPTION
**Pull request by ganik**

### CODE REVIEW

**Comment by justinormont** [comment]
@ganik: just pointing out that PR #100 updates to 1.0.0 of ML.NET; I'm hoping this isn't coincidence.

**Comment by ganik** [comment]
WIP: 27 tests to be fixed still

File: src/python/nimbusml.pyproj [hunk 1]
```diff
@@ -12,7 +12,7 @@
     <ProjectTypeGuids>{888888a0-9f3d-457c-b088-3a5042f75d52}</ProjectTypeGuids>
     <LaunchProvider>Standard Python launcher</LaunchProvider>
     <Name>nimbusml</Name>
-    <InterpreterId>Global|VisualStudio|Py3.7</InterpreterId>
+    <InterpreterId>Global|VisualStudio|MinePy37</InterpreterId>
```
Comment (shmoradims) [inline src/python/nimbusml.pyproj]: >MinePy37 [](start = 39, length = 8)

unintended change?

**Comment by shmoradims** [comment]
> FM = \

Did you double check that all the string replacements in this file are still working fine? If part of the old string has changed b/c of some renaming, the the replacement won't happen and you'll end up with a broken component. 

---
Refers to: src/python/tools/code_fixer.py:23 in 89afa9e. [](commit_id = 89afa9e468186e0eb55ab13860168fc1caa75d68, deletion_comment = False)

File: src/python/nimbusml/internal/core/linear_model/averagedperceptronbinaryclassifier.py [hunk 2]
```diff
@@ -107,27 +107,22 @@ class AveragedPerceptronBinaryClassifier(
...
 
-    :param num_iterations: Number of iterations.
+    :param number_of_iterations: Number of iterations.
 
-    :param init_wts_diameter: Sets the initial weights diameter that specifies
-        the range from which values are drawn for the initial weights. These
-        weights are initialized randomly from within this range. For example,
-        if the diameter is specified to be ``d``, then the weights are
-        uniformly distributed between ``-d/2`` and ``d/2``. The default value
-        is ``0``, which specifies that all the  weights are set to zero.
```
Comment (shmoradims) [inline src/python/nimbusml/internal/core/linear_model/averagedperceptronbinaryclassifier.py]: Looks like description for initial_weights_diameter is regressing to something useless. Could you please check why this is happening?

File: src/python/nimbusml/internal/core/linear_model/poissonregressionregressor.py [hunk 3]
```diff
@@ -62,43 +62,33 @@ class PoissonRegressionRegressor(
...
-        slower, more accurate.
+    :param optimization_tolerance: Tolerance parameter for optimization
+        convergence. Low = slower, more accurate.
 
-    :param memory_size: Memory size for L-BFGS. Lower=faster, less accurate.
-        The technique used for optimization here is L-BFGS, which uses only a
-        limited amount of memory to compute the next step direction. This
-        parameter indicates the number of past positions and gradients to store
-        for the computation of the next step. Must be greater than or equal to
-        ``1``.
```
Comment (shmoradims) [inline src/python/nimbusml/internal/core/linear_model/poissonregressionregressor.py]: same regression here

**Comment by shmoradims** [comment]
>             self._cv_kind = 'SignatureBinaryClassifierTrainer'

how come this is not changed like line 183 below?

---
Refers to: src/python/nimbusml/model_selection/cv.py:171 in 89afa9e. [](commit_id = 89afa9e468186e0eb55ab13860168fc1caa75d68, deletion_comment = False)

**Comment by shmoradims** [comment]
> # OneHotHashVectorizer

do we have a test that runs all the examples in src/python/nimbusml/examples/examples_from_dataframe/ and also in src/python/nimbusml/examples/examples_from_dataframe/examples_from_dataframe? I was expecting more changes here, given that a lot of the parameters were renamed/removed.

---
Refers to: src/python/nimbusml/examples/CountSelector.py:2 in 89afa9e. [](commit_id = 89afa9e468186e0eb55ab13860168fc1caa75d68, deletion_comment = False)

**Comment by ganik** [comment]
>             self._cv_kind = 'SignatureBinaryClassifierTrainer'

Agree, this is how its setup in ML.NET, small bug there

---
In reply to: [494120348](https://github.com/microsoft/NimbusML/pull/100#issuecomment-494120348) [](ancestors = 494120348)

---
Refers to: src/python/nimbusml/model_selection/cv.py:171 in 89afa9e. [](commit_id = 89afa9e468186e0eb55ab13860168fc1caa75d68, deletion_comment = False)

File: src/python/nimbusml/internal/core/linear_model/averagedperceptronbinaryclassifier.py (hunk 2 above)
Comment (ganik) [reply src/python/nimbusml/internal/core/linear_model/averagedperceptronbinaryclassifier.py]: Thanks good catch. Due to param renaming I had to update manifest_diff.json with new param names so the description is picked up

---
In reply to: [285733648](https://github.com/microsoft/NimbusML/pull/100#discussion_r285733648) [](ancestors = 285733648)

**Comment by ganik** [comment]
> # OneHotHashVectorizer

We do run all the examples from both folders. Even though there were many parameter name changes, in examples mostly they are not explicitly used

---
In reply to: [494124427](https://github.com/microsoft/NimbusML/pull/100#issuecomment-494124427) [](ancestors = 494124427)

---
Refers to: src/python/nimbusml/examples/CountSelector.py:2 in 89afa9e. [](commit_id = 89afa9e468186e0eb55ab13860168fc1caa75d68, deletion_comment = False)

**Comment by ganik** [comment]
> FM = \

Thank u, good catch. Double checked and cleaned up.

---
In reply to: [494070161](https://github.com/microsoft/NimbusML/pull/100#issuecomment-494070161) [](ancestors = 494070161)

---
Refers to: src/python/tools/code_fixer.py:23 in 89afa9e. [](commit_id = 89afa9e468186e0eb55ab13860168fc1caa75d68, deletion_comment = False)

**Pull request merged**